### PR TITLE
*: panic guard for warnings

### LINKED ISSUE #48756: runtime error shown in warning instead of truncate
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
CREATE TABLE t (id INT, a VARBINARY(20), b BIGINT);
INSERT INTO t VALUES(1, _binary '2012-05-19 09:06:07', 20120519090607),
(1, _binary '2012-05-19 09:06:07', 20120519090607),
(2, _binary '12012-05-19 09:06:07', 120120519090607),
(2, _binary '12012-05-19 09:06:07', 120120519090607);
SELECT SUBTIME(BIT_OR(b), '1 1:1:1.000002') FROM t GROUP BY id;
show warnings;
```

### 2. What did you expect to see? (Required)
Truncate warning
```
mysql> CREATE TABLE t (id INT, a VARBINARY(20), b BIGINT);
Query OK, 0 rows affected (0,02 sec)

mysql> INSERT INTO t VALUES(1, _binary '2012-05-19 09:06:07', 20120519090607),
    -> (1, _binary '2012-05-19 09:06:07', 20120519090607),
    -> (2, _binary '12012-05-19 09:06:07', 120120519090607),
    -> (2, _binary '12012-05-19 09:06:07', 120120519090607);
Query OK, 4 rows affected (0,00 sec)
Records: 4  Duplicates: 0  Warnings: 0

mysql> SELECT SUBTIME(BIT_OR(b), '1 1:1:1.000002') FROM t GROUP BY id;
+--------------------------------------+
| SUBTIME(BIT_OR(b), '1 1:1:1.000002') |
+--------------------------------------+
| 2012-05-18 08:05:05.999998           |
| NULL                                 |
+--------------------------------------+
2 rows in set, 1 warning (0,00 sec)

mysql> show warnings;
+---------+------+---------------------------------------------------+
| Level   | Code | Message                                           |
+---------+------+---------------------------------------------------+
| Warning | 1292 | Truncated incorrect time value: '120120519090607' |
+---------+------+---------------------------------------------------+
1 row in set (0,00 sec)
```
### 3. What did you see instead (Required)
ERROR 1105 (HY000): runtime error: invalid memory address or nil pointer dereference
```
tidb> CREATE TABLE t (id INT, a VARBINARY(20), b BIGINT);
Query OK, 0 rows affected (0.08 sec)

tidb> INSERT INTO t VALUES(1, _binary '2012-05-19 09:06:07', 20120519090607),
    -> (1, _binary '2012-05-19 09:06:07', 20120519090607),
    -> (2, _binary '12012-05-19 09:06:07', 120120519090607),
    -> (2, _binary '12012-05-19 09:06:07', 120120519090607);
Query OK, 4 rows affected (0.01 sec)
Records: 4  Duplicates: 0  Warnings: 0

tidb> SELECT SUBTIME(BIT_OR(b), '1 1:1:1.000002') FROM t GROUP BY id;
+--------------------------------------+
| SUBTIME(BIT_OR(b), '1 1:1:1.000002') |
+--------------------------------------+
| NULL                                 |
| 2012-05-18 08:05:05.999998           |
+--------------------------------------+
2 rows in set, 2 warnings (0.00 sec)

tidb> show warnings;
ERROR 1105 (HY000): runtime error: invalid memory address or nil pointer dereference
tidb> 
```
### 4. What is your TiDB version? (Required)

```
tidb_version(): Release Version: v7.6.0-alpha
Edition: Community
Git Commit Hash: 058a9614b22b42e1a93546e1f0e7ee1aae1ceb05
Git Branch: heads/refs/tags/v7.6.0-alpha
UTC Build Time: 2023-11-21 14:29:14
GoVersion: go1.21.3
Race Enabled: false
Check Table Before Drop: false
Store: tikv
```

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

Issue Number: close #48756

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
Ref scripts in #48756 
```
TiDB root@localhost:test> show warnings
+---------+------+-----------------------------------------+
| Level   | Code | Message                                 |
+---------+------+-----------------------------------------+
| Warning | 1292 | Incorrect time value: '120120519090607' |
| Warning | 1105 |                                         |
+---------+------+-----------------------------------------+
2 rows in set
Time: 0.004s
```
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/show.go [hunk 1]
```diff
@@ -1879,7 +1879,11 @@ func (e *ShowExec) fetchShowWarnings(errOnly bool) error {
 			sqlErr := terror.ToSQLError(x)
 			e.appendRow([]any{w.Level, int64(sqlErr.Code), sqlErr.Message})
 		default:
-			e.appendRow([]any{w.Level, int64(mysql.ErrUnknown), warn.Error()})
+			var err string
+			if warn != nil {
+				err = warn.Error()
+			}
+			e.appendRow([]any{w.Level, int64(mysql.ErrUnknown), err})
```
Comment (Defined2014) [inline pkg/executor/show.go:1886]: LGTM, but why is there an empty warning?

**Review by Defined2014 — APPROVED** [review]

File: pkg/executor/show.go (hunk 1 above)
Comment (xhebox) [reply pkg/executor/show.go:1886]: I don't know, maybe we can open a new issue? We should be able to track the source by 1105 code.

**Review by mjonss — COMMENTED** [review]
LGTM, but please also add a test case, to prevent regressions. I would suggest an integration test just with the statements from the reported issue.

**Review by mjonss — APPROVED** [review]
I added the hold label, so you can add a test, feel free to remove the label and merge it.

**Pull request merged**